src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskDeletionService.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.query.TaskResourceCaseQueryBuilder;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.TERMINATED;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.utils.DeleteTasksServiceHelper.getTaskIds;
+
+@Slf4j
+@Service
+public class TaskDeletionService {
+
+    private final CFTTaskDatabaseService cftTaskDatabaseService;
+
+    @Autowired
+    public TaskDeletionService(final CFTTaskDatabaseService cftTaskDatabaseService) {
+        this.cftTaskDatabaseService = cftTaskDatabaseService;
+    }
+
+    public void deleteTasksByCaseId(final String caseId) {
+        final List<TaskResourceCaseQueryBuilder> taskResourceCaseQueryBuilders = cftTaskDatabaseService
+                .findByTaskIdsByCaseId(caseId);
+
+        filterAllUnterminatedTasksAndLogError(taskResourceCaseQueryBuilders, caseId);
```

Comment:
As discussed, could this be updated to reflect that the "delete" logging in  filterAllUnterminatedTasksAndLogError happens after we call  deleteTasks(taskResourceCaseQueryBuilders, caseId);

Reply:
done

Reply:
Done